uk_election_timetables/election.py

```diff
@@ -99,3 +99,16 @@ def timetable(self) -> List[Dict]:
 
     def _calendar(self):
         return self.BANK_HOLIDAY_CALENDAR.from_country(self.country)
+
+    def get_date_for_event_type(self, event):
+        for e in self.timetable:
+            if e["event"] == event.name:
+                return e["date"]
+
+    def is_before(self, event):
```

Comment:
I think testing will be easier if we allow passing an optional date in to these methods, e.g:

```py
def is_before(self, event, date=None):
    if not date:
        date = datetime.now(timezone.utc).date()
    return self.get_date_for_event_type(event) > date
```